### ISSUE TITLE
Where is the git history? What is changed in code? How did the attack happen?

### ISSUE THREAD

**Issue by Altcoin-Cash** [bitcoinVBR/bitcoinV#1]

**Comment by bitcoinVBR** [comment]
Thanks for the questions.

There was an attack on BitcoinV, in particular the VBR feature was attacked. The attacker took advantage of the low hashrate on the BTCV blockchain and performed a 51% attack on it. On top of this, the attacker was able to set the VBR multiplier to the max level and pump out many rewards. Over the course of a week, the attacker mined about 800000 BTCV! This attack made it clear that a fix must be put in place. Since the attacker just about doubled the amount of coins mined in just a week, it was decided that rather hacking up the codebase to block his addresses and other UTXOs, we would start from scratch to keep the code clean and in line with the latest BTC release. 

The fix for the above attack was to only allow the VBR feature for difficulty levels above 16 Billion. Such a high difficulty level would suggest that the BTCV blockchain would be very mature and have many solo and pool miners on it making it incredibly difficult and extremely costly to perform the VBR attack on it. The VBR feature was also enabled for the first couple thousands of mined blocks. These rewards will be used for swapping all BTCV on all exchanges as well as pools. One million BTCV have been reserved for submitting BTCV to new exchanges in the future.

Another code update was to change from 48 block step diff, to a LWMA filtered approach for the difficulty adjustment. This allows mining to be consistent as the block diff updates every block. The problem with the previous approach was that one miner would mine the heck out of the chain and then stop; this left the difficulty high and miners typically waited for "others" to mine again to bring the diff down again.

As far as GIT history, the idea of BTCV is simple, always stay synced with the latest version of Bitcoin code. To see the diff, simply get your favorite diff utility (Beyond Compare is very nice) and diff with the latest version of Bitcoin:
                                        https://github.com/bitcoin/bitcoin/tree/v0.20.0

The above updates will never prevent a 51% attack, but it will eliminate exploiting the VBR feature because to exploit the chain now, one would have to spend at least 1000 times more than they previously did. This eliminates most of the community.and would suggest that only very large communities would have the ability to gang up and try to VBR attack the coin. If that ever happens, BTCV is doing very well to say the least!

**Comment by Altcoin-Cash** [comment]
Having a coin code without its history from conception is poor practice. It is not doable to look through the diff against Bitcoin and spot the latest VBR change from the previous VBR when the previous code is missing.

**Comment by laakie** [comment]
hold on, did we just restarted the chain from genesis ?

**Comment by bitcoinVBR** [comment]
> Having a coin code without its history from conception is poor practice. It is not doable to look through the diff against Bitcoin and spot the latest VBR change from the previous VBR when the previous code is missing.

I see your point. The diff is simply the versions of BTC with the corresponding version of BTCV. In this case we incremented a 1 to distinguish between this version and old versions out there.

BitcoinV is not a coin that will diverge and really need much of a codebase. In the future, the best practice will be to take the latest Bitcoin code and apply a patch file to it for the conversion into BTCV. We could just pull request a simple .patch file in the future.

**Comment by bitcoinVBR** [comment]
> hold on, did we just restarted the chain from genesis ?
> http://explorer.bitcoinv.org/block-height/2
> 
> Timestamp
> **2020-08-03 15:24:41 utc**

yes. A hacker almost doubled the amount of BTCV inside of 1 week. This was a big eye opener. A fix has been put in place. Rather than hacking up the existing code to try and prevent  his massive amount of BTCV from being spent, a new chain was formed.

**Comment by laakie** [comment]
> yes. A hacker almost doubled the amount of BTCV inside of 1 week. This was a big eye opener. A fix has been put in place. Rather than hacking up the existing code to try and prevent his massive amount of BTCV from being spent, a new chain was formed.

Maybe it's a good thing that happened then, at least we're now getting rid of the diff traps.
Just need to make sure everybody can swap their coins, 
My funds are on Mining-Dutch, and as far as I can see, their balance has gone to zero..

**Comment by Altcoin-Cash** [comment]
Of course, all balances go to zero since it is a new chain. There is no coins to swap with, your valueables went up in smoke.
Old nodes are still running old chain, with all coins.

**Comment by bitcoinVBR** [comment]
> > yes. A hacker almost doubled the amount of BTCV inside of 1 week. This was a big eye opener. A fix has been put in place. Rather than hacking up the existing code to try and prevent his massive amount of BTCV from being spent, a new chain was formed.
> 
> Maybe it's a good thing that happened then, at least we're now getting rid of the diff traps.
> Just need to make sure everybody can swap their coins,
> My funds are on Mining-Dutch, and as far as I can see, their balance has gone to zero..

Yes, all funds on all Exchanges and Pools will be swapped. On an individual basis people can reach out to Jay Medyaa. He will require some form of proof of coins, such as a screenshot of a wallet.

**Comment by Tillkoeln** [comment]
[https://twitter.com/TillKoeln/status/1293562800049197057?s=20](https://twitter.com/TillKoeln/status/1293562800049197057?s=20)

**Comment by bitcoinVBR** [comment]
yes, many have been mined early. About 1.5 million were used for swapping. Another million are being used for listing on Finexbox and a soon to be announced exchange and future exchange. Spreading BTCV to more exchanges is one way BTCV will continue to grow for the community.
